lib/active_merchant/billing/gateways/paypal_express.rb

```diff
@@ -146,7 +146,7 @@ def build_setup_request(action, money, options)
               xml.tag! 'n2:cpp-payflow-color', options[:background_color] unless options[:background_color].blank?
               if options[:allow_guest_checkout]
                 xml.tag! 'n2:SolutionType', 'Sole'
-                xml.tag! 'n2:LandingPage', options[:landing_page] || 'Billing'
+                xml.tag! 'n2:LandingPage', options[:landing_page] || 'Billing' unless options[:paypal_chooses_landing_page]
```

Comment:
Whatever the param, it's hard to understand what the behaviour of the code is because of the way it's written.